### PR TITLE
#2286: Remove `gcc-8` support

### LINKED ISSUE #2286: Remove support for `gcc-8`
**What Needs to be Done?**
https://github.com/DARMA-tasking/vt/pull/2282 utilizes `[[maybe_unused]]` to silence a bunch of warnings, but support for that attribute is incomplete in `gcc-8` (see [Slack discussion](https://darmatalk.slack.com/archives/C01SEK9R3M1/p1715802883068769)).

Remove support for `gcc-8` and bump compiler requirement to `gcc-9` for `vt`:
- remove `gcc-8` build from CI
- update documentation and wiki
- remove workarounds from b832ffc5532d9d43da9dbd4e9913fa5a8d12481a

### PR DESCRIPTION
**Pull request by thearusable**
Closes #2286

### CODE REVIEW

**Comment by cz4rs** [comment]
gcc9 + zoltan build has a couple of following:
`##[warning]Free memory is lower than 5%; Currently used: 96.92%`

This actually causes to build to fail even though build and tests complete fine. Investigating.

**Comment by cz4rs** [comment]
Should be good, macOS failure seems unrelated:
image

File: examples/lb_data/lb_data_file_example.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
         {
             "communications": [
                 {
-                    "bytes": 152.0,
+                    "bytes": 160.0,
```
Comment (cz4rs) [inline examples/lb_data/lb_data_file_example.json:26]: I assume this is a result of switching JSON validation from `gcc-9` to `clang-16`.

File: docker-compose.yml [hunk 2]
```diff
@@ -10,7 +10,7 @@
 # Variables:
 #   ARCH={amd64, arm64v8, ...}
 #   COMPILER_TYPE={gnu, clang, intel, nvidia}
-#   COMPILER={gcc-8, gcc-9, gcc-10,
+#   COMPILER={gcc-9, gcc-10,
```
[the remark this answers is not in the document]
Comment (cz4rs) [reply docker-compose.yml]: Yes, added some other missing variants too.

File: README.md [hunk 3]
```diff
@@ -4,7 +4,6 @@
 [![codecov](https://codecov.io/gh/DARMA-tasking/vt/branch/develop/graph/badge.svg)](https://codecov.io/gh/DARMA-tasking/vt)
 [![License](https://img.shields.io/badge/License-BSD%203--Clause-orange.svg)](https://opensource.org/licenses/BSD-3-Clause)
 ![](https://github.com/DARMA-tasking/vt/workflows/Docker%20Image%20CI/badge.svg)
-[![gcc-8, ubuntu, mpich, address sanitizer](https://dev.azure.com/DARMA-tasking/DARMA/_apis/build/status/PR%20tests%20(gcc-8%2C%20ubuntu%2C%20mpich%2C%20address%20sanitizer)?branchName=develop&Label=(gcc-8%2C%20ubuntu%2C%20mpich%2C%20address%20sanitizer))](https://dev.azure.com/DARMA-tasking/DARMA/_build/latest?definitionId=9&branchName=develop)
```
[the remark this answers is not in the document]
Comment (cz4rs) [reply README.md:7]: Done.

File: ci/docker/ubuntu-clang-cpp.dockerfile [hunk 4]
```diff
@@ -64,52 +66,9 @@ ENV MPI_EXTRA_FLAGS="" \
...
-    VT_PRODUCTION_BUILD_ENABLED=${VT_PRODUCTION_BUILD_ENABLED} \
-    VT_TRACE_ENABLED=${VT_TRACE_ENABLED} \
-    VT_TRACE_RUNTIME_ENABLED=${VT_TRACE_RUNTIME} \
-    VT_UBSAN_ENABLED=${VT_UBSAN_ENABLED} \
-    VT_UNITY_BUILD_ENABLED=${VT_UNITY_BUILD_ENABLED} \
-    VT_WERROR_ENABLED=${VT_WERROR_ENABLED} \
-    CMAKE_CXX_STANDARD=${CMAKE_CXX_STANDARD} \
-    VT_DEBUG_VERBOSE=${VT_DEBUG_VERBOSE} \
-    VT_CI_BUILD=${VT_CI_BUILD}
-
```
Comment (cz4rs) [inline ci/docker/ubuntu-clang-cpp.dockerfile:109]: Drive-by change: those are not necessary to pass environment variables.

**Review by cwschilly — APPROVED** [review]
LGTM

File: src/vt/collective/reduce/scoping/strong_types.h [hunk 5]
```diff
@@ -101,7 +101,12 @@ struct TagPair {
   }
 
   TagType first() const { return t1_; }
+#pragma GCC diagnostic push  // ignore gcc-13 false positive
+#ifndef __clang__  // clang does not recognize "-Wmaybe-uninitialized"
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#endif
   TagType second() const { return t2_; }
+#pragma GCC diagnostic pop
```
Comment (PhilMiller) [inline src/vt/collective/reduce/scoping/strong_types.h:109]: This is really strange, and possibly a compiler bug that should be reduced and reported. I can't see any way that `t2_` could be uninitialized, regardless of the constructor used, or any other possible constructor, since it has a default initializer.

Are you sure the warning isn't actually coming from something else nearby a use of this, and this inadvertently suppresses it when there's something real to be fixed?

Comment (cz4rs) [reply src/vt/collective/reduce/scoping/strong_types.h:109]: I've spent some time scratching my head over this. As weird as it is, the warning seemed to come directly from this line.
I will give it another spin tomorrow to make sure.

Comment (cz4rs) [reply src/vt/collective/reduce/scoping/strong_types.h:109]:
```
root@c43073bf8b2d:/build/vt# /usr/bin/g++ -DJSON_USE_IMPLICIT_CONVERSIONS=1 -DVT_NO_COLOR_ENABLED -I/vt/lib/CLI -I/build/vt/release -I/vt/src -I/vt/lib/json/include -I/vt/lib/brotli/c/include -I/vt/lib/libfort/lib -I/build/vt/lib/checkpoint/src -I/vt/lib/checkpoint/src -isystem /vt/lib/fmt/include -isystem /vt/lib/EngFormat-Cpp/include -isystem /vt/lib/yaml-cpp/include -O3 -DNDEBUG -Wall -Wextra -Wno-unknown-pragmas -Wnon-virtual-dtor -Wshadow -Wsign-compare -Wsuggest-override -pedantic -ftemplate-backtrace-limit=100 -Werror -fsanitize=address -fno-omit-frame-pointer -fno-optimize-sibling-calls -std=c++17 -o src/CMakeFiles/vt.dir/vt/collective/collective_scope.cc.o -c /vt/src/vt/collective/collective_scope.cc
In file included from /vt/src/vt/collective/reduce/reduce_scope.h:47,
                 from /vt/src/vt/collective/reduce/reduce_manager.h:47,
                 from /vt/src/vt/collective/collective_alg.h:52,
                 from /vt/src/vt/collective/collective_scope.cc:49:
In member function 'vt::TagType vt::collective::reduce::detail::TagPair::second() const',
    inlined from 'bool vt::collective::reduce::detail::TagPair::operator==(const vt::collective::reduce::detail::TagPair&) const' at /vt/src/vt/collective/reduce/scoping/strong_types.h:91:47,
    inlined from 'std::operator==<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>(const variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&, const variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&)::<lambda(auto:14&&, auto:15)> mutable [with auto:14 = const vt::collective::reduce::detail::TagPair&; auto:15 = std::integral_constant<long unsigned int, 1>]' at /usr/include/c++/13/variant:1266:3,
    inlined from 'constexpr _Res std::__invoke_impl(__invoke_other, _Fn&&, _Args&& ...) [with _Res = void; _Fn = operator==<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>(const variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&, const variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&)::<lambda(auto:14&&, auto:15)>; _Args = {const vt::collective::reduce::detail::TagPair&, integral_constant<long unsigned int, 1>}]' at /usr/include/c++/13/bits/invoke.h:61:36,
    inlined from 'constexpr typename std::__invoke_result<_Functor, _ArgTypes>::type std::__invoke(_Callable&&, _Args&& ...) [with _Callable = operator==<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>(const variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&, const variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&)::<lambda(auto:14&&, auto:15)>; _Args = {const vt::collective::reduce::detail::TagPair&, integral_constant<long unsigned int, 1>}]' at /usr/include/c++/13/bits/invoke.h:96:40,
    inlined from 'static constexpr decltype(auto) std::__detail::__variant::__gen_vtable_impl<std::__detail::__variant::_Multi_array<_Result_type (*)(_Visitor, _Variants ...)>, std::integer_sequence<long unsigned int, __indices ...> >::__visit_invoke(_Visitor&&, _Variants ...) [with _Result_type = std::__detail::__variant::__variant_idx_cookie; _Visitor = std::operator==<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>(const variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&, const variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&)::<lambda(auto:14&&, auto:15)>&&; _Variants = {const std::variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&}; long unsigned int ...__indices = {1}]' at /usr/include/c++/13/variant:1049:17,
    inlined from 'constexpr decltype(auto) std::__do_visit(_Visitor&&, _Variants&& ...) [with _Result_type = __detail::__variant::__variant_idx_cookie; _Visitor = operator==<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>(const variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&, const variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&)::<lambda(auto:14&&, auto:15)>; _Variants = {const variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&}]' at /usr/include/c++/13/variant:1816:5,
    inlined from 'constexpr decltype(auto) std::__do_visit(_Visitor&&, _Variants&& ...) [with _Result_type = __detail::__variant::__variant_idx_cookie; _Visitor = operator==<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>(const variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&, const variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&)::<lambda(auto:14&&, auto:15)>; _Variants = {const variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&}]' at /usr/include/c++/13/variant:1756:5,
    inlined from 'constexpr void std::__detail::__variant::__raw_idx_visit(_Visitor&&, _Variants&& ...) [with _Visitor = std::operator==<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>(const variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&, const variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&)::<lambda(auto:14&&, auto:15)>; _Variants = {const std::variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>&}]' at /usr/include/c++/13/variant:183:44,
    inlined from 'constexpr bool std::operator==(const variant<_Types ...>&, const variant<_Types ...>&) [with _Types = {vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType}]' at /usr/include/c++/13/variant:1266:3,
    inlined from 'vt::collective::reduce::detail::ReduceStamp vt::collective::reduce::Reduce::reduceImmediate(vt::NodeType, MsgT*, vt::collective::reduce::detail::ReduceStamp, ReduceNumType) [with MsgT = vt::collective::CollectiveAlg::CollectiveMsg; void (* f)(MsgT*) = vt::collective::reduce::operators::ReduceCombine<>::msgHandler<vt::collective::CollectiveAlg::CollectiveMsg, vt::collective::reduce::operators::None<char>, vt::collective::reduce::operators::NoCombine>]' at /vt/src/vt/collective/reduce/reduce.impl.h:168:20:
/vt/src/vt/collective/reduce/scoping/strong_types.h:104:35: error: '*(const vt::collective::reduce::detail::TagPair*)((char*)&<unnamed> + offsetof(std::variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>,std::variant<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>::<unnamed>.std::__detail::__variant::_Variant_base<vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>::<unnamed>.std::__detail::__variant::_Move_assign_base<true, vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>::<unnamed>.std::__detail::__variant::_Copy_assign_base<true, vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>::<unnamed>.std::__detail::__variant::_Move_ctor_base<true, vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>::<unnamed>.std::__detail::__variant::_Copy_ctor_base<true, vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>::<unnamed>.std::__detail::__variant::_Variant_storage<true, vt::util::strong::detail::Strong<int, -1, vt::collective::reduce::detail::tags::TagTag>, vt::collective::reduce::detail::TagPair, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::SeqTag>, vt::util::strong::detail::Strong<long unsigned int, 18446744073709551615, vt::collective::reduce::detail::tags::UserIDTag>, vt::epoch::EpochType>::_M_u)).vt::collective::reduce::detail::TagPair::t2_' may be used uninitialized [-Werror=maybe-uninitialized]
  104 |   TagType second() const { return t2_; }
      |                                   ^~~
In file included from /vt/src/vt/collective/reduce/reduce.h:543,
                 from /vt/src/vt/collective/reduce/reduce_manager.impl.h:48,
                 from /vt/src/vt/collective/collective_alg.h:173:
/vt/src/vt/collective/reduce/reduce.impl.h: In member function 'vt::collective::reduce::detail::ReduceStamp vt::collective::reduce::Reduce::reduceImmediate(vt::NodeType, MsgT*, vt::collective::reduce::detail::ReduceStamp, ReduceNumType) [with MsgT = vt::collective::CollectiveAlg::CollectiveMsg; void (* f)(MsgT*) = vt::collective::reduce::operators::ReduceCombine<>::msgHandler<vt::collective::CollectiveAlg::CollectiveMsg, vt::collective::reduce::operators::None<char>, vt::collective::reduce::operators::NoCombine>]':
/vt/src/vt/collective/reduce/reduce.impl.h:168:31: note: '<anonymous>' declared here
  168 |   auto cur_id = id == detail::ReduceStamp{} ? generateNextID() : id;
      |                               ^~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

Comment (cz4rs) [reply src/vt/collective/reduce/scoping/strong_types.h]: It seems like the issue here is caused by passing the temporary to `TagPair::operator==` (nothing unusual about it as far as I can see). b4925858b2a6821876c20e9aa3a5b1d0e5ed0801 is just a workaround with a local variable.

My manual attempts at reduced replication were not successful, so I'll try to plug this into C-Reduce whenever I have some spare time.

File: examples/lb_data/lb_data_file_example.json (hunk 1 above)
Comment (cz4rs) [reply examples/lb_data/lb_data_file_example.json:26]: @lifflander
This has a simpler explanation :facepalm: `clang-16` build has `VT_TRACE: 1` and `gcc-9` build has `VT_TRACE: 0`.
This is consistent with what I see locally, there's no difference between gcc and clang as long as they are build with the same setting.

**Review by lifflander — APPROVED** [review]
